src/paint.js

```diff
@@ -129,12 +129,26 @@ function convertHYPERCUBEtoJSON(layout) {
     var dim1Id = qMatrix.map(function(d) {
       return d[0].qElemNumber;
     });
+    var dim1IsNull = qMatrix.map(function(d) {
+      if (d[0].qIsNull){
+        return d[0].qIsNull;
+      }else{
+        return false;
+      }
+    });
     var dim2Labels = qMatrix.map(function(d) {
       return d[1].qText;
     });
     var dim2Id = qMatrix.map(function(d) {
       return d[1].qElemNumber;
     });
+    var dim2IsNull = qMatrix.map(function(d) {
```

Comment:
figure out a more suitable name for these, dim2IsNull suggests it has to do with whether or not the dimension is null whereas this is an index or lookup table for whether or not a specific entry is null.